src/JSInterop/Microsoft.JSInterop/src/DotNetObjectRefJsonConverterFactory.cs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.JSInterop
+{
+    internal sealed class DotNetObjectReferenceJsonConverterFactory : JsonConverterFactory
+    {
+        public override bool CanConvert(Type typeToConvert)
+        {
+            Debug.Assert(
+                typeToConvert.IsGenericType && typeToConvert.GetGenericTypeDefinition() == typeof(DotNetObjectRef<>),
+                "We expect this to only be called for DotNetObjectRef<T> instances.");
+            return true;
```

Comment:
This kind of thing is broken. It won't actually throw an exception or prevent this from happening. It would be better to just write the implementation correctly (return false for unsupported).